### ISSUE TITLE
Can't login as Admin to MediaWiki

### ISSUE THREAD

**Issue by Sepero** [NixOS/nixpkgs#290618]
### Describe the bug
MediaWiki installs, and loads, and appears to run properly. But I'm unable to login. I can't figure out how to login. Looking the code below, the install script, the administrator username appears to be "admin".
https://github.com/NixOS/nixpkgs/blob/526d051b128b82ae045a70e5ff1adf8e6dafa560/nixos/modules/services/web-apps/mediawiki.nix#L620

I have modified my code to match that of the MediaWiki test, yet I still cannot login.

```
{ config, pkgs, lib, ... }:

{
  services.mediawiki = {
    enable = true;
    httpd.virtualHost = {
      hostName = "localhost";
      adminAddr = "root@example.com";
    };
    passwordFile = pkgs.writeText "password" "correcthorsebatterystaple";
    name = "Sample Wiki";
    webserver = "nginx";
  };
}
```

I have tried logging in a "admin", "user", and "mediawiki".

### Steps To Reproduce
1. Install MediaWiki service using test code (above).
2. Attempt to login with "admin" + "correcthorsebatterystaple"
3. Error: Incorrect username or password

### Notify maintainers
@Mic92 @pennae @h7x4

**Comment by Mic92** [comment]
Notice the option says "initial" password. Are you using an existing database?

**Comment by Sepero** [comment]
@Mic92 Thank you for the quick reply. I'm not using a pre-existing database, but I did have troubles with the password initially. If using an initial password of 1 character length, then MediaWiki seems to be completely broken. So I put the Nix configuration that matches the Nix module test (as shown in OP). Also, I have attempted to delete the entire database and rebuild/reinstall several times (`sudo rm -rf /var/lib/mysql`). Still I can't login.

**Comment by Sepero** [comment]
I'm just checking in to see if there's any update on getting this working.

**Comment by Sepero** [comment]
Someone able to test and verify whether I'm missing a step or anything?

**Comment by Sepero** [comment]
Today I tested the exact same configuration, but added channel **21.11 as nixpkgs2111**, tested that old package

    service.mediawiki.package = nixpkgs2111.mediawiki;

Attempt to login with user **admin** password **correcthorsebatterystaple**

Same result as OP. No difference, can't login.

**Comment by Sepero** [comment]
![Screenshot_20240312-135821](https://github.com/NixOS/nixpkgs/assets/1575546/a403c89c-418a-435f-98aa-c0eff46e4c1b)

**Comment by Sepero** [comment]
Note:
I deleted all of /var/lib/{mediawiki,mysql} before running this test again.

**Comment by Sepero** [comment]
With help from a random user on Matrix, we have discovered that this is simply another case of poor NixOS documentation. I'm very disappointed that this issue has gone on for over a month (and could have continued for how many more months?)

I know the devs are busy with many other things, but if someone knowledgeable would have kindly taken 10 minutes to simply install MediaWiki, then this could have been resolved very quickly. Instead, more time than that was wasted with Message replies between myself and @Mic92, and my project postponed for weeks. I'm a Nix newbie, but based on unfounded speculation, I took an hour out of my day to rule out the possibility of Regression. For what reason?? It makes no sense. **All the information needed to reproduce issue was in the OP.**

As stated in the OP

> I have tried logging in a "admin", "user", and "mediawiki".

THE SOLUTION:
Someone saying: Oh ye, hey you know that admin account? That's not admin, haha. The admin is "Wiki"

WTF- Wiki

As stated in the OP

> I have tried logging in a "admin", "user", and "mediawiki".

Hopefully my frustration is coming through without offense. There is no indication of admin named "Wiki" to be found anywhere. How could such a simple problem have failed to resolve on so many levels?

1. Not only did I fail to get help here.
2. Failure on NixOS Wiki. https://nixos.wiki/wiki/MediaWiki
3. Failure of Documentation.
4. Failure of community. As this solution has been solved by other NixOS users before, yet no one has reported it to Devs, nor corrected it on the NixOS Wiki. This issue has existed for how many years? Since at least 2021.

Credit goes to the community on NixOS Matrix chat. I got the solution there. REALLY smart and helpful people are on Matrix who answer many questions daily... But unfortunately, Matrix chat is ephemeral.

TO CONCLUDE:

While I'm very disappointed, I don't want to neglect all the incredible work all Devs are doing (and with little compensation). I'm very thankful for everything you do. You are true heroes in a world fought with non-free software. Without your devotion, NixOS and Linux would not exist.

I have a request. Please be careful not to take those of us who report bugs for granted. You know 99% of people never report any bugs. I'm in that 1% who often reports bugs. I'm not a developer. I'm not a Nix guru. Collecting proper information for you takes me a good deal of time, and I always try to write good reports with all necessary information. When you make a request of me to provide you more information, then I have to try to recreate the exact scenario to reproduce the problem, then add in those extra details you request. It takes a dummy like me a lot of time. A lot more time than it would take a expert such as yourself.

**Comment by JulienMalka** [comment]
As you said wisely, nixpkgs maintainers are often overworked and with very little time to answer to such requests. The right place to find help is indeed matrix. Please, if you find the time, submit a PR to improve mediawiki's documentation in nixpkgs!

**Comment by Sepero** [comment]
Understood. Thank you

**Comment by threddast** [comment]
Thank you for going through this, I had the same problem. 
However, I'm a bit confused because I just installed the version of Mediawiki included with Nixos 23.11 and the admin account is "MediaWiki", not "Wiki"
So, which one is it?

**Comment by Sepero** [comment]
Cheers. Essentially the issue is, Don't use spaces in the MediaWiki name prior to 24.05, Else it will parse incorrectly. The administrator login name should be "admin"
